packages/lit-dev-content/site/docs/localization/transform-mode.md

```diff
@@ -0,0 +1,285 @@
+example below is safe because a script can only be loaded if it matches one of
+our known locale codes, but if our matching logic was less precise, it could
+result in bugs or attacks that inject insecure JavaScript.
+
+</div>
+
+```js
+import {allLocales} from './generated/locales.js';
+
+const url = new URL(window.location.href);
+const unsafeLocale = url.searchParams.get('locale');
+const locale = allLocales.includes(unsafeLocale) ? unsafeLocale : 'en';
+
+const script = document.createElement('script');
+script.type = 'module';
+script.src = `/${locale}.js`;
+document.head.appendChild(script);
+```
+
+Implementing logic similar to this on your server so that the appropriate script
+tag is statically rendered into your HTML file will usually result in the best
+performance, because the browser will start downloading your script as early as
+possible.
+
+## Switching locales
+
+In transform mode, the `setLocale` function is not available. Instead, reload
+the page so that the next load will pick a different locale bundle.
+
+For example, this `locale-picker` custom element loads a new URL whenever a new
+locale is selected from a drop-down list:
+
+{% switchable-sample %}
+
+```ts
+import {LitElement, html} from 'lit';
+import {customElement} from 'lit/decorators.js';
+import {getLocale} from './localization.js';
+import {allLocales} from './generated/locales.js';
+
+@customElement('locale-picker');
+export class LocalePicker extends LitElement {
+  render() {
+    return html`
+      <select @change=${this.localeChanged}>
+        ${allLocales.map(
+          (locale) =>
+            html`<option value=${locale} selected=${locale === getLocale()}>
+              ${locale}
+            </option>`
+        )}
+      </select>
+    `;
+  }
+
+  localeChanged(event: Event) {
+    const newLocale = (event.target as HTMLSelectElement).value;
+    const url = new URL(window.location.href);
+    if (url.searchParams.get('locale') !== newLocale) {
+      url.searchParams.set('locale', newLocale);
+      window.location.assign(url.href);
+    }
+  }
+}
+```
+
+```js
+import {LitElement, html} from 'lit';
+import {getLocale} from './localization.js';
+import {allLocales} from './generated/locales.js';
+
+export class LocalePicker extends LitElement {
+  render() {
+    return html`
+      <select @change=${this.localeChanged}>
+        ${allLocales.map(
+          (locale) =>
+            html`<option value=${locale} selected=${locale === getLocale()}>
+              ${locale}
+            </option>`
+        )}
+      </select>
+    `;
+  }
+
+  localeChanged(event) {
+    const newLocale = event.target.value;
+    const url = new URL(window.location.href);
+    if (url.searchParams.get('locale') !== newLocale) {
+      url.searchParams.set('locale', newLocale);
+      window.location.assign(url.href);
+    }
+  }
+}
+customElements.define('locale-picker', LocalePicker);
+```
+
+{% endswitchable-sample %}
+
+## Rollup
```

Comment:
Maybe "Rollup integration"? Trying to find some kind of alternate phrasing that makes it clearer when you look at the TOC or search results that you're talking about an optional integration.

Reply:
Done